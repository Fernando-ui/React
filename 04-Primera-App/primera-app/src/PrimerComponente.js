import PropTypes from 'prop-types';
const PrimerComponente = ({saludo, subtitulo}) => {

    return (
    <>
    <h1>{saludo}</h1>
    <h1>{subtitulo}</h1>
    </>)

}

PrimerComponente.propTypes = {

    saludo:PropTypes.string.isRequired,
    
};

PrimerComponente.defaultProps = {

    subtitulo:'Soy un subtitulo'
}
export default PrimerComponente;
