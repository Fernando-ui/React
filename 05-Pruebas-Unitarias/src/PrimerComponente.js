import PropTypes from 'prop-types';
const PrimerComponente = ({saludo, subtitulo}) => {

    return (
    <>
    <h1>{saludo}1</h1>
    <h1>{subtitulo}</h1>
    </>)

}

PrimerComponente.propTypes = {

    saludo:PropTypes.string.isRequired,
    
};

PrimerComponente.defaultProps = {

    subtitulo:'Soy un subtitulo'
}
export default PrimerComponente;
