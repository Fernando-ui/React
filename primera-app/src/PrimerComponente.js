const PrimerComponente = () => {

    const saludo = 'Hola desde una variable';
    const obje = {
        nombre:'fer',
        apellido:'sosa'
    }

    return (
    <>
    <h1>Hola amigos{saludo}</h1>
    <pre>Adios amigos{JSON.stringify(obje,null,3)} </pre>
    </>)

}
export default PrimerComponente
