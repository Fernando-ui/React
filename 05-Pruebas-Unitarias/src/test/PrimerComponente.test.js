import React from 'react';
import {shallow, configure} from 'enzyme';
import PrimerComponente from "../PrimerComponente";
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import '@testing-library/jest-dom';
import {createSerializer} from 'enzyme-to-json';

expect.addSnapshotSerializer(createSerializer({mode: 'deep'}));

configure({adapter: new Adapter()})
describe('Pruebas en <PrimerComponente />', () => {

    
    test('Debe de mostrar  <PrimerComponente /> correctamente', () => {
        const saludo = 'Hola, Soy Fernando';
        const wrapper = shallow(<PrimerComponente saludo={saludo}/>);
        
        expect(wrapper).toMatchSnapshot();
    });
    
});
