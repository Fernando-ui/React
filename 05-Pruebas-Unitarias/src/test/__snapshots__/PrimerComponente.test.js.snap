// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <PrimerComponente /> Debe de mostrar  <PrimerComponente /> correctamente 1`] = `
<Fragment>
  <h1>
    Hola, Soy Fernando
    1
  </h1>
  <h1>
    Soy un subtitulo
  </h1>
</Fragment>
`;
